backend/controllers/otpController.js

```diff
@@ -0,0 +1,83 @@
+const nodemailer = require("nodemailer");
+const Brand = require("../model/brands");
+const OTP = require("../model/otp");
+const dbConfig = require("../config/dbConfig");
+
+const transporter = nodemailer.createTransport({
+  host: dbConfig.SMTP_HOST,
+  port: 465,
+  secure: true,
+  auth: {
+    user: dbConfig.OTP_MAIL,
+    pass: dbConfig.PASSWORD,
+  },
+});
+function generateOTP() {
+  return Math.floor(100000 + Math.random() * 900000).toString();
+}
+
+const sendOTP = async (req, res) => {
+  const { email } = req.body;
+
+  // Check if email exists
+  //Uncomment this only after integrating the brand with database
+  //let brand = await Brand.findOne({ email });
+  // if (!brand) {
+  //     return res.status(400).send('Email does not exist');
+  // }
+  let _email = await OTP.findOne({ email });
+  if (_email) {
+    if (Date.now() < _email.otpExpires) {
+      return res.status(400).send("OTP is not expired yet");
+    }
+  }
+  // Generate OTP
+  const otp = generateOTP();
+  const otpExpires = Date.now() + 3600000;
+  if (_email) {
+    _email.otp = otp;
+    _email.otpExpires = otpExpires;
+    await _email.save();
+  } else {
+    const newOTP = new OTP({
+      email,
+      otp,
+      otpExpires,
+    });
+    await newOTP.save();
+  }
+  const mailOptions = {
+    from: dbConfig.OTP_MAIL,
+    to: email,
+    subject: "Verify Your OTP Code",
```

Comment:
lets create a `constant` folder and have `constants` file inside it containing all the hardcoded things like error message, this one etc so that it becomes easier to handle

Reply:
ok